plugin-annotation/src/main/java/com/mapbox/maps/plugin/annotation/AnnotationManagerImpl.kt

```diff
@@ -185,6 +181,19 @@ abstract class AnnotationManagerImpl<G : Geometry, T : Annotation<G>, S : Annota
     }
 
     updateSource()
+
+    // Listen to map change event to reload layer and source in the first init
+    if (!mapChangeListenerAdded) {
+      mapChangeListenerAdded = true
+      delegateProvider.mapListenerDelegate.addOnMapChangedListener {
```

Comment:
Should this listener be cleared as others in `onDestroy`? If yes - then flag should be reset in `onDestroy` as well.